server/src/main/java/org/opensearch/cluster/NodeConnectionsService.java

```diff
@@ -159,6 +164,14 @@ public void connectToNodes(DiscoveryNodes discoveryNodes, Runnable onCompletion)
         runnables.forEach(Runnable::run);
     }
 
+    public void setPendingDisconnections(Set<DiscoveryNode> nodes) {
+        nodes.forEach(transportService::setPendingDisconnection);
+    }
+
+    public void clearPendingDisconnections() {
+        transportService.clearPendingDisconnections();
+    }
```

Comment:
Can we add UT's for these new methods ?
We need to verify that the ConnectionChecker is able to handle the error being thrown by TransportService#connectToNode . Lets add tests to `NodeConnectionsServiceTests`

Reply:
added a test to verify that ConnectionChecker is able to keep running in the background and eventually succeed when pending disconnects is cleared